### ISSUE TITLE
ElementSiteTreeFilterSearch very resource intensive

### ISSUE THREAD

**Issue by HJGreen** [dnadesign/silverstripe-elemental#801]
## Problem

When using the CMS SiteTree search on a site with Elemental installed, the response can be extremely slow.

I've tracked it down to this method, which it seems renders **every** Element on **every** Page to a template in order to perform a text search on the contents.

https://github.com/dnadesign/silverstripe-elemental/blob/4/src/Controllers/ElementSiteTreeFilterSearch.php#L44

For smaller sites this isn't a big deal but the performance drops very quickly as the number of Pages and Elements grow. 

In our case it got even worse as we had some `[embed][/embed]` shortcodes and so for each one of those the server was sending a HTTP request to YouTube. This resulted in a whole bunch of rate limiting and request timeouts.

## Temporary Solution

We've disabled the `ElementSiteTreeFilterSearch` extension via the config YAML files using:
```yml
---
Name: cmssearch
After: '#elemental'
---
SilverStripe\Core\Injector\Injector:
  SilverStripe\CMS\Controllers\CMSSiteTreeFilter_Search:
    class: SilverStripe\CMS\Controllers\CMSSiteTreeFilter_Search
  DNADesign\Elemental\Controllers\ElementSiteTreeFilterSearch:
    class: SilverStripe\CMS\Controllers\CMSSiteTreeFilter_Search

```

I'm happy to help with a resolution to this, but would like to hear your ideas for this as well. My thoughts would be to:

1. Disable the extension by default, or provide a simpler way to disable it initially
2. Look into optimising the search method to avoid rendering to templates if possible

**Comment by michalkleiner** [comment]
@maxime-rainville a candidate for the performance regression testing perhaps?

**Comment by robbieaverill** [comment]
Duplicate of https://github.com/dnadesign/silverstripe-elemental/issues/189
